### ISSUE TITLE
Update starter repos

### ISSUE THREAD

**Issue by LucyGwilliamAdmin** [open-sdg/open-sdg#1093]
@LucyGwilliamAdmin @brockfanning @phillipgardiner thought we could capture any changes below

- [x] 2020 indicator changes
- [x] split out metadata/indicator config into two folders
- [x] only provide minimal data and metadata e.g. one of each for 1.1.1
- [x] rename "_prose.yml" to something more descriptive, like "metadata_schema.yml"? (after the merging of #1082 I think we can officially drop any mention of Prose.io)
- [x] Split out site config fields into _data/site_config.yml
- [x] Update FAQ page - [Issue with suggested changes](https://github.com/open-sdg/open-sdg-site-starter/issues/51)
- [x] update About page content
- [x] guidance page - is this needed? Could just use content in FAQs and remove Guidance
- [x] Link to Contact page in footer
- [x] Update production setup

**Comment by brockfanning** [comment]
Added an item above about the _prose.yml file in the data starter.

**Comment by LucyGwilliamAdmin** [comment]
@brockfanning @AnnCorp just had a thought about changing the repo names to data and site - I'm now wondering if we should go ahead with that since it's referred to in so many places e.g. written docs, video tutorials.

What are your thoughts?

I don't think it would stop us from recommending/using the naming convention suggested? i.e. sdg-[COUNTRY]/site and sdg-[COUNTRY]/data?

**Comment by AnnCorp** [comment]
@LucyGwilliamAdmin no objection from me if no downside to it, but @brockfanning's view more needed than mine on this one.

**Comment by brockfanning** [comment]
@LucyGwilliamAdmin Sounds like you're thinking we should not change the names? I can see the benefit there - since they are already referred to as "open-sdg-site-starter" and "open-sdg-data-starter", like you say. Also, it can be nice to have a descriptive name of a repository. So I'd be fine with not changing them.

**Comment by LucyGwilliamAdmin** [comment]
@brockfanning @anncorp ok, thanks - I'm going to remove that item from the list then

**Comment by LucyGwilliamAdmin** [comment]
@brockfanning I've opened PR for renaming _prose.yml:

https://github.com/open-sdg/open-sdg-data-starter/pull/49

**Comment by LucyGwilliamAdmin** [comment]
@AnnCorp @brockfanning Re. pages, does anyone have any objection to removing the guidance page and maybe reincorporating some of the info into the other pages (if not already)?

**Comment by AnnCorp** [comment]
@LucyGwilliamAdmin am definitely in favour of removing guidance page and seeing where else the info could be put, if still required.

**Comment by LucyGwilliamAdmin** [comment]
@brockfanning I've just added an item for splitting out the site config fields into _data/site_config.yml - does that sound good with you?

**Comment by brockfanning** [comment]
@LucyGwilliamAdmin This sounds good to me. But yes I think this would mean an update to the quickstart. But I think this could be a significant improvement. Here's my thinking:

* I like how the data starter is operational immediately, without any steps. Users just click "use this template" and (once the build completes) they have a working data repo. It would be great if the same could be done for the site repo...
* The one obstacle to achieving this with the site repo is that "baseurl" part. So I'm thinking we should update the quickstart such that users are _required_ to name their site repo specifically "open-sdg-site-starter". This would mean that -- just like the data repo -- users just click "use this template" and (once the build completes) they have a working site repo.
* Of course we do want users to be able to customize the name of their site repo, so we could add an optional section at the end for renaming it, which would also include a description of updating the "baseurl".
* If we're now going to recommend/document the site config form, then we could update the "remote_data_prefix" part to describe that process using the site config form, instead of directly editing a file.
* Also, if we are going to now recommend/document the use of the metadata and data config forms, then we could remove the part in the quickstart about "data_edit_url" and "metadata_edit_url". (Since the "Edit data" and "Edit metadata" buttons would just be pointing at the built-in forms.)

I'm interested to hear your thoughts on this. If it would help I could rewrite the written quickstart to show exactly how it would be different.

**Comment by LucyGwilliamAdmin** [comment]
I really like the sound of both repos working without any changes being needed so support both of those things. I know we spoke about recommending having 'data' and 'site' for the repo names but maybe we could just include that recommendation in the optional section, especially now we're not going to change the template names?

The only other thing I'm think of is the quick start video but I imagine that wouldn't take too long to edit the small changes in - @AnnCorp does that sound ok to you?

**Comment by AnnCorp** [comment]
@LucyGwilliamAdmin yes, sounds good to me - thanks.

**Comment by brockfanning** [comment]
@LucyGwilliamAdmin I see that Github now automatically disables workflows that include the "scheduled" option. They can still be enabled, but it means an extra step for the quick start. Do you think we should just remove the "scheduled" from that workflow, with the assumption that countries will set up the automatic site builds?

**Comment by LucyGwilliamAdmin** [comment]
@brockfanning hmm, good point. I think if we go down that route, will need to say that either they will need to set up automatic site builds or rebuild the site build manually for changes to data repo to apply.

Although, that being said - I created a new site today and didn't need to enable any workflows so not sure if something I did during set up?

**Comment by brockfanning** [comment]
Hmm ok, just wondering - have you tried making a change to a file? And it triggered the "deploy to staging" workflow?

**Comment by LucyGwilliamAdmin** [comment]
@brockfanning I've just done a quick test, just creating the repo and not changing anything and it seems the 'Deploy to staging' workflow is enabled: https://github.com/LucyGwilliamAdmin/open-sdg-data-starter/actions

**Comment by brockfanning** [comment]
Ok, maybe it was just a fluke or caused by some other factor. Just to be sure can you try making a commit directly to the develop branch? If that builds normally then I it may not be a problem.

**Comment by LucyGwilliamAdmin** [comment]
@brockfanning so I was testing with data but just realised that it doesn't have a scheduled workflow so just tested with site.

Without doing anything other than setting up the repo from template, I get this on Actions tab - it says that the workflow is disable but still runs and then once the action finished, the message goes

![image](https://user-images.githubusercontent.com/43404358/121510225-4e17ba80-c9df-11eb-9df2-8468320d5c6d.png)

**Comment by brockfanning** [comment]
@LucyGwilliamAdmin To be sure, can you make an additional commit directly to the develop branch?

**Comment by LucyGwilliamAdmin** [comment]
@brockfanning oh yeah, that doesn't trigger a build. However a site does get built from the initial deployment so we would still be able to achieve getting a site set up without any changes being needed so maybe we could put in enabling the workflow as a next step?

**Comment by brockfanning** [comment]
Ok, yep I guess the question is, do we:
a) Add a step to the quickstart to enable the workflow, or
b) Remove the schedule from the site starter and then add instructions (maybe [here](https://open-sdg.readthedocs.io/en/latest/automation/triggered-site-builds/)) on adding the schedule

Any preferences?

**Comment by AnnCorp** [comment]
@brockfanning @LucyGwilliamAdmin don't have enough understanding of the tech side of things to get if we think having the schedule in site starter is a better thing than not having it - do the pros outweigh the cons? Your thoughts/votes?

**Comment by brockfanning** [comment]
With the schedule, if someone makes a change to their data repository, after 24 hours that change will be reflected on the site, automatically.

Without the schedule, if someone makes a change to their data repository, it will not be reflected on the site until they make some change (any change) on the site repository. This means that if they are unaware that they need to do this, then the site may not be updated for a very long time.

However with the "triggered site builds" procedure that Lucy did, if someone makes a change to their data repository, it will be reflected on the site within about 10 minutes.

My feeling is that the triggered site build has been a critical feature for users. I don't think that the 24 hour wait is acceptable for most users. We added that 24-hour schedule as a fallback, to make sure that there is some automated way to get the data changes onto the site (eventually), particularly for the benefit of non-technical users who may not be aware that they need to trigger a site build in order to see their data changes. But I think the triggered site build is a *huge* improvement. (I can't imagine doing any of the Open SDG trainings I've done without the triggered site builds.)

So I guess my vote would be to remove the schedule and maybe mention it as a possibility in the docs on setting up the triggered site builds.

**Comment by LucyGwilliamAdmin** [comment]
@brockfanning that all sounds good to me - I'd just think we need to be clear about them needing to set up triggered builds

**Comment by brockfanning** [comment]
@LucyGwilliamAdmin I've put up a PR #1230 with some proposed documentation updates, which itself depends on PRs in jekyll-open-sdg-plugins and open-sdg-site-starter. It uses environment variables in Github Actions to simplify the quickstart. Happy to talk through it on a tech call or answer any questions. It also handles the "split out config fields" mentioned here.

**Comment by LucyGwilliamAdmin** [comment]
Re. linking to Contact us page in footer, I'm just revisiting the [UK Contact us page](https://sdgdata.gov.uk/contact-us/) and it actually not really possible to make it general without making it compulsory to edit.

I think we could still add a link to the existing contact us page that we provide and improve it, just not move to UK template.

What are others thoughts?
@phillipgardiner @AnnCorp @brockfanning

**Comment by LucyGwilliamAdmin** [comment]
@brockfanning can you think of anything that is left for starter repos this release?

**Comment by brockfanning** [comment]
@LucyGwilliamAdmin I think there may be some updates we need to do in regards to the production setup. I'm testing it out now.

**Comment by LucyGwilliamAdmin** [comment]
@brockfanning think this can be closed right?
